requests/utils.py

```diff
@@ -447,6 +446,36 @@ def get_encodings_from_content(content):
             xml_re.findall(content))
 
 
+def _parse_content_type_header(header):
+    """Returns content type and parameters from given header
+
+    :param header: string
+    :return: tuple containing content type and dictionary of
+         parameters
+    """
+    if not header:
+        return None
+    # append delimiter on end to ensure at least two elements when split by ';'
+    header += ';'
```

Comment:
I don't think this is necessary. Can we do

```py
tokens = header.split(';', 1)
content_type, params_list = tokens[0].strip(), tokens[1:]
if  len(params_list) < 1:
    return content_type, {}
params = params_list[0]
```

If I understand correctly that's the effect of this otherwise. It's simple, clearer, and doesn't require us to string concatenation which results in a new allocation and isn't terribly efficient.

```py
>>> a = [1]
>>> b, c = a[0], a[1:]
>>> b, c
(1, [])
```

Reply:
That is pretty nice.  Didn't know a[1:] would not throw an index out of bounds... nice.